src/core/algorithms/ind/mind/mind.cpp

```diff
@@ -0,0 +1,271 @@
+/** \file
+ * \brief Spider algorithm
+ *
+ * Spider algorithm class methods definition
+ */
+#include "mind.h"
+
+#include <algorithm>
+#include <unordered_set>
+
+#include "algorithms/create_algorithm.h"
+#include "config/error/option.h"
+#include "config/names_and_descriptions.h"
+#include "ind/ind_algorithm.h"
+#include "max_arity/option.h"
+#include "table/column_combination.h"
+#include "table/dataset_stream_fixed.h"
+#include "table/dataset_stream_projection.h"
+#include "tabular_data/input_table_type.h"
+#include "timed_invoke.h"
+
+namespace algos {
+
+Mind::Mind() : INDAlgorithm({}) {
+    MakeLoadOptsAvailable();
+
+    RegisterOption(config::kErrorOpt(&max_ind_error_));
+    RegisterOption(config::kMaxArityOpt(&max_arity_));
+}
+
+void Mind::MakeLoadOptsAvailable() {
+    /*
+     * At the moment we only have one algorithm for mining unary approximate inds.
+     * In the future we should give the user the ability to choose the algorithm.
+     */
+    auind_algo_ = CreateAlgorithmInstance<INDAlgorithm>(AlgorithmType::spider);
+}
+
+void Mind::MakeExecuteOptsAvailable() {
+    MakeOptionsAvailable({config::kErrorOpt.GetName(), config::kMaxArityOpt.GetName()});
+}
+
+bool Mind::SetExternalOption(std::string_view option_name, boost::any const& value) {
+    try {
+        auind_algo_->SetOption(option_name, value);
+        return true;
+    } catch (config::ConfigurationError&) {
```

Comment:
Should we handle this error somehow? Or just return false?

Reply:
This function is needed to try setting the option for the unary algorithm. If it doesn’t work out, then we’ll return the false here and then (in the `Algorithm::SetOption`) try to set it up for the mind. If it doesn’t work there, we’ll throw a configuration error to the user.
At the same time, we call this function for all passed options, including those that may be specific to the `Mind`. In this case, when trying to set it for a unary algorithm, we may get an error that we should simply suppress.